internal/controller/controllers/local_cluster_import_controller_test.go

```diff
@@ -0,0 +1,371 @@
+
+	var verifyClusterDeploymentNotPresent = func() {
+		err := getClusterDeployment()
+		Expect(err).To(HaveOccurred())
+		Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+	}
+
+	var verifyAgentClusterInstallNotPresent = func() {
+		err := getAgentClusterInstall()
+		Expect(err).To(HaveOccurred())
+		Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+	}
+
+	var deleteAgentServiceConfig = func() {
+		err := client.Delete(ctx, agentServiceConfig)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	var deleteManagedCluster = func() {
+		err := client.Delete(ctx, managedCluster)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	BeforeEach(func() {
+		// Create a context for the test:
+		ctx = context.Background()
+		scheme := runtime.NewScheme()
+		utilruntime.Must(clusterv1.AddToScheme(scheme))
+		utilruntime.Must(configv1.AddToScheme(scheme))
+		utilruntime.Must(hivev1.AddToScheme(scheme))
+		utilruntime.Must(hiveext.AddToScheme(scheme))
+		utilruntime.Must(corev1.AddToScheme(scheme))
+		utilruntime.Must(aiv1beta1.AddToScheme(scheme))
+		client = fakeclient.NewClientBuilder().WithScheme(scheme).Build()
+
+		//Create objects to be reconciled / watched.
+		agentServiceConfig = &aiv1beta1.AgentServiceConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "agent",
+			},
+			Spec: aiv1beta1.AgentServiceConfigSpec{
+				DatabaseStorage: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Ti"),
+						},
+					},
+				},
+				FileSystemStorage: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Ti"),
+						},
+					},
+				},
+				ImageStorage: &corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Ti"),
+						},
+					},
+				},
+			},
+		}
+		key = clnt.ObjectKeyFromObject(agentServiceConfig)
+		managedCluster = &clusterv1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "local-cluster",
+			},
+			Spec: clusterv1.ManagedClusterSpec{
+				HubAcceptsClient:     true,
+				LeaseDurationSeconds: 60,
+			},
+		}
+		request = ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Name: agentServiceConfig.Name,
+			},
+		}
+
+		// Create the reconciler:
+		reconciler = NewLocalClusterImportReconciler(client, "local-cluster", "agent", logrusLogger)
+	})
+
+	It("should create cluster import CRs when ManagedCluster and AgentServiceConfig are present", func() {
+		createManagedCluster()
+		createAgentServiceConfig()
+		createNodes()
```

Comment:
There's a way to "seed" the fake client with some initial data so you don't have to do all this boiler plate each time. Check it out in the agentserviceconfig tests: https://github.com/openshift/assisted-service/blob/17959954d606e1faac51baa251d5df3e090dd3f9/internal/controller/controllers/agentserviceconfig_controller_test.go#L49-L58

Then I think it would be good to not hide all the implementation in functions.